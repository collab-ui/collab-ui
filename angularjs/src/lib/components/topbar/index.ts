import * as angular from 'angular';
import { TopbarComponent } from './topbar.component';

export default angular
  .module('momentum.ui.topbar', [])
  .component('mdTopBar', new TopbarComponent())
  .name;
