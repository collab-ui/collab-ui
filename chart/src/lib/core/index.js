const RegUpperLetter = /^[A-Z]/;
const ModifyApplyProperty = {
  'style': 1,
  'attr': 1,
  'classed': 1,
  'property': 1,
  'text': 1,
  'html': 1
};

const core = {
  guid: function () {
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
      let r = Math.random() * 16 | 0;
      let v = c === 'x' ? r : (r & 0x3 | 0x8);
      return v.toString(16);
    });
  },
  // to be used or removed
  access: function (elem, key, value, fn) {
    var length = elem.length;
    // fake bool
    if (typeof length === 'number') {
      for (var i = 0; i < length; i++) {
        fn(elem[i], key, value);
      }
    } else {
      fn(elem, key, value);
    }
  },
  isArray: function (obj) {
    return Object.prototype.toString.call(obj) === '[object Array]';
  },
  isArguments: function (obj) {
    return Object.prototype.toString.call(obj) === '[object Arguments]';
  },
  // add('ad',1,2) add([])
  // core.extendArguments(arguments, this._remove, this);
  extendArguments: function (args, fn, caller) {
    if (args.length && args.length > 0 && core.isArray(args[0])) {
      const _argList = args[0];
      let _returns = [];
      for (var i = 0, l = _argList.length; i < l; i++) {
        if (core.isArray(_argList[i])) {
          _returns.push(fn.apply(caller, _argList[i]));
        } else {
          _returns.push(fn.call(caller, _argList[i]));
        }
      }
      return _returns;
    } else {
      return fn.apply(caller, args);
    }
  },
  lowerCase: function (word) {
    return word.toLocaleLowerCase();
  },
  lowerCaseInital: function (word) {
    return word.replace(RegUpperLetter, core.lowerCase);
  },
  modifySelection: function (selection, config) {
    let args;
    for (let prop in config) {
      if (ModifyApplyProperty[prop]) {
        args = config[prop];
        for (let n in args) {
          selection[prop](n, args[n]);
        }
      }
    }
  },
  assignUnKnownLv2: function (target, source) {
    for (let prop in source) {
      if (target[prop] === undefined) {
        target[prop] = Object.assign({}, source[prop]);
      } else {
        Object.assign(target[prop], source[prop]);
      }
    }
  },
  getReturnOrValue: function (config, d, i) {
    if (typeof config === 'function') {
      return config(d, i);
    } else {
      return config;
    }
  },
  getFirstValue: function () {
    for (let i = 0, l = arguments.length; i < l; i++) {
      if (arguments[i] !== undefined) {
        return arguments[i];
      }
    }
    return undefined;
  },
  callInner: function (target, config, ifResetTarget) {
    for (let name in config) {
      if (typeof target[name] === 'function') {
        if (ifResetTarget) {
          target = target[name](config[name]);
        } else {
          target[name](config[name]);
        }
      }
    }
    return target;
  },
  applyInner: function (target, config, caller) {
    for (let name in config) {
      if (typeof target[name] === 'function') {
        target[name].apply(caller, config[name]);
      }
    }
    return target;
  }
};

export default core;
