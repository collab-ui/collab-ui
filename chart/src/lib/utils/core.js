const RegUpperLetter = /^[A-Z]/;
const ModifyApplyProperty = {
  'style': 1,
  'attr': 1,
  'classed': 1,
  'property': 1,
  'text': 1,
  'html': 1
};

const core = {
  create: function (tag) {
    if (process && process.env && process.env.NODE_ENV === 'test') {
      return document.createElement(tag);
    } else {
      return document.createElementNS("http://www.w3.org/2000/svg", tag);
    }
  },
  template: function (str, obj) {
    let reg;
    for (let n in obj) {
      reg = new RegExp('\\$' + n + '\\$', 'g');
      str = str.replace(reg, obj[n]);
    }
    return str;
  },
  guid: function () {
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
      let r = Math.random() * 16 | 0;
      let v = c === 'x' ? r : (r & 0x3 | 0x8);
      return v.toString(16);
    });
  },
  // to be used or removed
  access: function (elem, key, value, fn) {
    var length = elem.length;
    // fake bool
    if (typeof length === 'number') {
      for (var i = 0; i < length; i++) {
        fn(elem[i], key, value);
      }
    } else {
      fn(elem, key, value);
    }
  },
  toArray: function (obj) {
    return Array.prototype.slice.call(obj);
  },
  isArray: function (obj) {
    return Object.prototype.toString.call(obj) === '[object Array]';
  },
  isArguments: function (obj) {
    return Object.prototype.toString.call(obj) === '[object Arguments]';
  },
  // add('ad',1,2) add([['ad',1,2],['ad',1,2]])
  // core.extendArguments(arguments, this._remove, this);
  extendArguments: function (args, fn, caller) {
    if (args.length && args.length > 0 && core.isArray(args[0])) {
      return this.extendArray(args[0], fn, caller);
    } else {
      return fn.apply(caller, args);
    }
  },
  // add([1,2,3]) add(1)
  extendOneArgument: function (args, fn, caller) {
    if (args.length && args.length > 0 && core.isArray(args[0])) {
      return this.extendArray(args[0], fn, caller);
    } else {
      return fn.call(caller, args);
    }
  },
  applyOrCall: function (args, fn, caller) {
    return core.isArray(args) ? fn.apply(caller, args) : fn.call(caller, args);
  },
  extendArray: function (arr, fn, caller) {
    const _returns = [];
    for (var i = 0, l = arr.length; i < l; i++) {
      _returns.push(core.applyOrCall(arr[i], fn, caller));
    }
    return _returns;
  },
  lowerCase: function (word) {
    return word.toLocaleLowerCase();
  },
  lowerCaseInital: function (word) {
    return word.replace(RegUpperLetter, core.lowerCase);
  },
  modifySelection: function (selection, config) {
    let args;
    for (let prop in config) {
      if (ModifyApplyProperty[prop]) {
        args = config[prop];
        for (let n in args) {
          selection[prop](n, args[n]);
        }
      }
    }
  },
  assignUnKnownLv2: function (target, source) {
    for (let prop in source) {
      if (target[prop] === undefined) {
        target[prop] = Object.assign({}, source[prop]);
      } else {
        Object.assign(target[prop], source[prop]);
      }
    }
  },
  getReturnOrValue: function (config, d, i) {
    if (typeof config === 'function') {
      return config(d, i);
    } else {
      return config;
    }
  },
  getFirstValue: function () {
    for (let i = 0, l = arguments.length; i < l; i++) {
      if (arguments[i] !== undefined) {
        return arguments[i];
      }
    }
    return undefined;
  },
  callInner: function (target, config, ifResetTarget) {
    for (let name in config) {
      if (typeof target[name] === 'function') {
        if (ifResetTarget) {
          target = target[name](config[name]);
        } else {
          target[name](config[name]);
        }
      }
    }
    return target;
  },
  applyInner: function (target, config, caller) {
    for (let name in config) {
      if (typeof target[name] === 'function') {
        target[name].apply(caller, config[name]);
      }
    }
    return target;
  }
};

export default core;
