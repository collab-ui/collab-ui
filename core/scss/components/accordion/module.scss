@import '../../tools/functions/core';
@import '../../colors/settings';
@import '../../settings/core';
@import '../../settings/focus';
@import '../../settings/media';
@import '../../typography/settings';
@import '../list-item/settings';
@import '../../tools/mixins/core';
@import '~@momentum-ui/icons/scss/mixins';

@import 'settings';
@import 'mixins';
@import 'accordion';
