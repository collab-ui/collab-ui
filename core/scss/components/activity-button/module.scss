@import '../../tools/functions/core';
@import '../../colors/settings';
@import '../../settings/core';
@import '../button/settings';
@import '../button/mixins';

@import 'activity-button';
