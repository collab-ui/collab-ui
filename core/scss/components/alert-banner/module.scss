@import '../../settings/core';

@import 'settings';
@import 'alert-banner';
