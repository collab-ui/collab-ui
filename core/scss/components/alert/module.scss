@import '../../colors/settings';
@import '../../settings/core';
@import '../../settings/media';
@import '../../typography/settings';
@import '../icon/settings';
@import '../list-item/settings';
@import '~@momentum-ui/icons/scss/mixins';
@import '../../tools/mixins/flex';
@import '../../tools/mixins/border-radius';

@import 'settings';
@import 'mixins';
@import 'alert';
