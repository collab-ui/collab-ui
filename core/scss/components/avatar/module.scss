@import '../../tools/functions/core';
@import '~@momentum-ui/icons/scss/variables';
@import '../../colors/settings';
@import '../../settings/core';
@import '../../settings/media';
@import '../../typography/settings';
@import '../../settings/z-index';
@import 'settings';

@import '../../tools/mixins/core';
@import '../../tools/mixins/border-radius';
@import '../../tools/mixins/flex';
@import '../avatar/settings';
@import '../../colors/settings';
@import '../../settings/core';
@import 'mixins';

@import 'avatar';
@import 'composite-avatar';
