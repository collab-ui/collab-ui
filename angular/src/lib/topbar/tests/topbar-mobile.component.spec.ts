import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { TopbarModule } from '../topbar.module';

describe('TopbarMobileComponent', () => {
  @Component({
    selector: 'test-app',
    template: `
      <cui-top-bar-mobile>
        <div class="testingforTbM"></div>
      </cui-top-bar-mobile>
    `,
  })
  class TestAppComponent { }

  let testComponent: TestAppComponent;
  let fixture: ComponentFixture<TestAppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [TopbarModule],
      declarations: [TestAppComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestAppComponent);
    testComponent = fixture.componentInstance;
  });

  it('should create and match snapshot', () => {
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one TopbarMobile', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('cui-top-bar-mobile');
    expect(element.querySelector('.cui-top-bar__mobile')).not.toBeNull();
  });

  it('should render children', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.cui-tb-mobile__nav');
    expect(element.children[0].className).toContain('testingforTbM');
  });
});
