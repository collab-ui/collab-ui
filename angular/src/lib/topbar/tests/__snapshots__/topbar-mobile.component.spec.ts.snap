// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarMobileComponent should create and match snapshot 1`] = `
<test-app>
  <cui-top-bar-mobile>
    
    <cui-icon
      aria-pressed="false"
      buttonclassname="cui-top-bar__mobile-menu-button"
      name="list-menu_20"
      ng-reflect-aria-label="Open Menu"
      ng-reflect-button-class-name="cui-top-bar__mobile-menu-butto"
      ng-reflect-name="list-menu_20"
    >
      
      
      <button
        aria-label="getAriaLabel()"
        aria-labelledby="getAriaLabelledBy()"
        class="cui-button cui-button--icon cui-top-bar__mobile-menu-button"
      >
        <i
          class="cui-icon icon icon-list-menu_20"
        />
      </button>
    </cui-icon>
    <div
      class="cui-top-bar__mobile cui-tb-mobile"
      ng-reflect-ng-class="cui-top-bar__mobile,cui-tb-mob"
      role="menu"
      tabindex="0"
    >
      <cui-icon
        aria-pressed="false"
        buttonclassname="cui-tb-mobile__close"
        name="cancel_20"
        ng-reflect-aria-label="Close Menu"
        ng-reflect-button-class-name="cui-tb-mobile__close"
        ng-reflect-name="cancel_20"
      >
        
        
        <button
          aria-label="getAriaLabel()"
          aria-labelledby="getAriaLabelledBy()"
          class="cui-button cui-button--icon cui-tb-mobile__close"
        >
          <i
            class="cui-icon icon icon-cancel_20"
          />
        </button>
      </cui-icon>
      <span>
        <div
          class="cui-top-bar__brand"
        />
      </span>
      <cui-list-separator
        backgroundcolor="transparent"
        class="cui-list-separator"
        ng-reflect-background-color="transparent"
        style="background-color: transparent;"
      >
        <span
          class="cui-list-separator__container"
        >
          
          
        </span>
        
      </cui-list-separator>
      <nav
        class="cui-tb-mobile__nav"
      >
        <div
          class="testingforTbM"
        />
      </nav>
    </div>
    <div
      class="cui-tb-mobile__mask"
      ng-reflect-ng-class="cui-tb-mobile__mask,"
      role="none"
    />
  </cui-top-bar-mobile>
</test-app>
`;
